### PR TITLE
functions for parsing ENV variables as an array

### PR DESCRIPTION
**Pull request by psladek**
usage e.g: 
MY_ADV_OPTIONS=opt1,opt2,opt2

.. 
if (check_var_array('MY_ADV_OPTIONS',$opt)) {
 ...do setup option $opt
}

### CODE REVIEW

File: testapi.pm [hunk 1]
```diff
@@ -267,6 +267,21 @@ sub check_var($$) {
...
+    return $default if !@vars;
+    return \@vars;
+}
+
+sub check_var_array($$) {
+    my ($var, $val) = @_;
+    my $vars_r = get_var_array($var);
+    my %vars_h = map { $_ => 1 } @{$vars_r};
+    return 1 if (defined $vars_h{$val});
+    return 0;
```
Comment (coolo) [inline testapi.pm]: return defined $vars_h{$val}); ?

Comment (coolo) [inline testapi.pm]: and actually the map is useless too - you can just return grep { $_ eq $val } @$vars_r

**Comment by coolo** [comment]
please squash the commits and prepare a PR to openQA - where we unfortunately keep the API documentation

File: testapi.pm (hunk 1 above)
Comment (psladek) [inline testapi.pm]: hm, was writting it probably too much in inspiration to the 'check_var' function, no problem to simplify it like this :)

**Comment by psladek** [comment]
ok, more simplification and squash

**Pull request merged**